dask/dataframe/tests/test_dataframe.py

```diff
@@ -4135,6 +4135,23 @@ def test_set_index_with_index():
     assert ddf2 is ddf
 
 
+def test_set_index_no_sort():
```

Comment:
I'm not sure what best practices on the `dask/dask` side are, but I think this test would be more robust to changes if we didn't test for `len(result.__dask_graph__().layers)`. Instead, I suggest creating an (explicitly) unsorted `pd.Dataframe` as input and tested that the end result has the correct index set and is (un-)sorted.